### ISSUE TITLE
Suggestion: Parameterized default mapper for typescript-resolvers

### ISSUE THREAD

**Issue by jonaskello** [dotansimha/graphql-code-generator#1591]
**Is your feature request related to a problem? Please describe.**

Using the [default mapper](https://graphql-code-generator.com/docs/plugins/typescript-resolvers#defaultmapper-string) I cannot refer to the type that is being mapped.

**Describe the solution you'd like**

Today we can do this:

```yaml
plugins
  config:
    defaultMapper: any
```

From what I understand this feature can only replace all types by a single type (`any` in the example above).  I would like to be able to do this:

```yaml
plugins
  config:
    defaultMapper: Partial<{T}>
```

In this example `{T}` would get replaced with the type being mapped. For example if `User` is being mapped then it would be `Partial<User>`. This way I could make all types partial without losing type-safety. 

It could also be used to map types to an indirection. For example if I have this:

```ts
export type Parents = {
  User: UserSpecial;
  // Other parents
};
```

I could then do this:

```yaml
plugins
  config:
    defaultMapper: Parents["{T}"]
```

The benefit of this would be that I could refer to `Parents["User"]` and then easily change what this refers to (this would be another way to achieve what I suggested in #1589).

**Comment by dotansimha** [comment]
Awesome! I'll try to add it :)

**Comment by dotansimha** [comment]
WIP here: https://github.com/dotansimha/graphql-code-generator/pull/1593

**Comment by dotansimha** [comment]
Implemented in 1.0.7 :)
